### PR TITLE
Added the ability to set the background color

### PR DESCRIPTION
**Pull request by PGMacDesign**
Added 2 methods for setting the background color of the Elastic view. It should take in either an int color or a String and parse the color. If the string is null or the parsing fails, nothing will be set and it will retain the original color

### CODE REVIEW

File: elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java [hunk 1]
```diff
@@ -107,5 +107,33 @@ public void onEnterAnimationFinished() {
 
         // Do further actions if necessary
     }
+    
+    
+    /////////////////////////////////////////////////////////////////////////////
```
Comment (Tibolte) [inline elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java]: You can remove that comment, it's not really necessary

File: elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java [hunk 2]
```diff
@@ -107,5 +107,33 @@ public void onEnterAnimationFinished() {
...
+     *                    Color.parseColor(string)
+     */
+    public void setBackgroundColor(String passedColor){
+        if(passedColor == null){
+            return;
+        }
+        try {
+            int color = Color.parseColor(passedColor);
+            this.setBackgroundColor(color);
+        } catch (Exception e){}
```
Comment (Tibolte) [inline elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java]: [Don't catch generic exception](https://source.android.com/source/code-style.html#dont-catch-generic-exception)

File: elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java [hunk 3]
```diff
@@ -107,5 +107,33 @@ public void onEnterAnimationFinished() {
...
+    
+    
+    /////////////////////////////////////////////////////////////////////////////
+    ///Added on 2016-07-29 by PGMacDesign as per Issue #27 in Github/////////////
+    /////////////////////////////////////////////////////////////////////////////
+    /**
+     * Set the background color of the Elastic Download View
+     * @param passedColor int Color Color to set the background
+     */
+    public void setBackgroundColor(int passedColor){
```
Comment (Tibolte) [inline elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java]: Take a look at how the rest of the file is formatted (I'm talking about braces here), there's always one space between one parenthesis and any curly brace. Try to keep some consistency on the code styling :)

**Comment by PGMacDesign** [comment]
Thanks for the tips. Were you wanting me to change the code prior to merge? Or were you just offering advice?

**Comment by Tibolte** [comment]
That's better if you change it, it should be quick to do it ;)

**Comment by PGMacDesign** [comment]
Updated with said fixes. Let me know if you need anything else changed on there, I will be happy to do it.

File: elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java [hunk 4]
```diff
@@ -107,5 +107,30 @@ public void onEnterAnimationFinished() {
...
+        this.mProgressDownloadView.setBackgroundColor(mBackgroundColor);
+    }
+
+    /**
+     * Overloaded method, takes in a String color to parse
+     * @param passedColor String of a color hex color (IE: #fd5c79) that can be parsed by
+     *                    Color.parseColor(string)
+     */
+    public void setBackgroundColor(String passedColor) {
+        if(passedColor == null){
```
Comment (Tibolte) [inline elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java]: this `if` needs a space between itself and the parenthesis (like the rest of the code).

File: elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java [hunk 5]
```diff
@@ -107,5 +107,30 @@ public void onEnterAnimationFinished() {
...
+            return;
+        }
+        try {
+            int color = Color.parseColor(passedColor);
+            this.setBackgroundColor(color);
+        } catch (IllegalArgumentException e) {}
```
Comment (Tibolte) [inline elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java]: Better! What we usually do in the case is use `e.printstacktrace();` to notify in logcat that something when wrong, or you can use a log message with `Log.e`. Either way we should do something here, so developers can get notified when a wrong color has been passed to this function.

File: elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java (hunk 4 above)
Comment (Tibolte) [inline elasticdownload/src/main/java/is/arontibo/library/ElasticDownloadView.java]: Just missing `if (passedColor == null)` (adding a white space) and it's good to go.

**Pull request merged**